plugins/scaffolder-backend/src/scaffolder/actions/builtin/github/githubTemplateRepoCreate.ts

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { InputError } from '@backstage/errors';
+import {
+  GithubCredentialsProvider,
+  ScmIntegrationRegistry,
+} from '@backstage/integration';
+import { Octokit } from 'octokit';
+import { createTemplateAction } from '../../createTemplateAction';
+import { parseRepoUrl } from '../publish/util';
+import {
+  createGithubTemplateRepoWithCollaboratorsAndTopics,
+  getOctokitOptions,
+} from './helpers';
+import * as inputProps from './inputProperties';
+import * as outputProps from './outputProperties';
+
+/**
+ * Creates a new action that initializes a git repository
+ *
+ * @public
+ */
+export function createGithubTemplateRepoCreateAction(options: {
+  integrations: ScmIntegrationRegistry;
+  githubCredentialsProvider?: GithubCredentialsProvider;
+}) {
+  const { integrations, githubCredentialsProvider } = options;
+
+  return createTemplateAction<{
+    repoUrl: string;
```

Comment:
Consider adding support for the following configuration options (similar to repo:create publish action):
```suggestion
    repoUrl: string;
    defaultBranch?: string;
    deleteBranchOnMerge?: boolean;
    allowRebaseMerge?: boolean;
    allowSquashMerge?: boolean;
    allowMergeCommit?: boolean;
    requireCodeOwnerReviews?: boolean;
    requiredStatusCheckContexts?: string[];
    topics?: string[];
    configureBranchProtection: boolean;
```